app/waf/src/controllers/adc.controller.ts

```diff
@@ -193,4 +207,143 @@ export class AdcController {
       );
     }
   }
+
+  // TODO: schema.response not work well here.
+  // It shows the below in Example Value:
+  // {
+  //   "actionsresponse": {
+  //     "id": "11111111-2222-3333-4444-555555555555"
+  //   }
+  // }
+  @post(prefix + '/adcs/{adcId}/action', {
+    responses: {
+      '200': Schema.response(ActionsResponse, 'Adc Id for the actions.'),
+    },
+  })
+  async provision(
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) id: string,
+    @requestBody(Schema.createRequest(ActionsRequest, 'actions request'))
+    actionBody: ActionsBody,
+  ): Promise<object | undefined> {
+    let adc = await this.adcRepository.findById(id);
+
+    switch (Object.keys(actionBody)[0]) {
+      case 'create':
+        try {
+          if (adc.status !== 'NONE' && adc.status !== 'ERROR')
+            throw new HttpErrors.BadRequest(
+              'Adc status is ' +
+                adc.status +
+                ". Cannot repeat 'create' on the same ADC.",
+            );
+
+          // TODO: Create VM in async way.
+          return await this.createOn(adc).then(async () => {
+            return {id: adc.id};
+          });
+        } catch (error) {
+          throw new HttpErrors.BadRequest(error.message);
+        }
+
+      case 'delete':
```

Comment:
add something in future？

Reply:
Yeah, we will implement delete and setup in the future.